### ISSUE TITLE
[LEAGUE LOADER]: AttributeError: 'Matchup' object has no attribute 'winner_team_key'

### ISSUE THREAD

**Issue by tprusisz** [joeyagreco/leeger#42]
### League Loader

Yahoo

### Summary

Thanks so much for this program. I don't have much Python experience, so that might be my issue. But I am very interested in trying to get this to work. It looks like there is lots of good info. Anyways, I successfully logged into the Yahoo Fantasy API (I gave it permission). But now I am getting that error when trying to run this line: 
league: League = yahooLeagueLoader.loadLeague()

I am trying to run this for a current league (2022). Maybe that is the issue? The full error I think is this:
teamAHasTiebreaker = yahooMatchup.winner_team_key == yahooTeamA.team_key and yahooMatchup.is_tied == 0
AttributeError: 'Matchup' object has no attribute 'winner_team_key'

Maybe only having data for Week 1 is not enough for this program to do everything it needs to do?

### Error Logs

teamAHasTiebreaker = yahooMatchup.winner_team_key == yahooTeamA.team_key and yahooMatchup.is_tied == 0
AttributeError: 'Matchup' object has no attribute 'winner_team_key'

**Comment by joeyagreco** [comment]
This seems to be a duplicate of [this issue](https://github.com/joeyagreco/leeger/issues/40), which there is now an untested fix out for.

It is in [version 1.2.0](https://github.com/joeyagreco/leeger/releases/tag/v1.2.0)

If you run `pip install --upgrade leeger` and then run the script again this bug _should_ be resolved.

**Comment by joeyagreco** [comment]
@tprusisz is this working for you now? Just checking if that's why you closed it.

**Comment by joeyagreco** [comment]
Can you send the line of code you have calling the YahooLeagueLoader?

**Comment by tprusisz** [comment]
Sure thing. I tried copying it from the example you have:

```
from leeger.league_loader import YahooLeagueLoader
from leeger.model.league import League

clientId = "XXXXX"
clientSecret = "XXX"
#yahooLeagueLoader.login(clientId=clientId, clientSecret=clientSecret)
yahooLeagueLoader = YahooLeagueLoader("XXXXXX", [2022], clientId=clientId, clientSecret=clientSecret,
                                      loginTimeoutSeconds=4)
league: League = yahooLeagueLoader.loadLeague()
```

And then I call it from the command prompt (using Windows) with:
`py yahooLeagueLoaderExample2.py`

**Comment by tprusisz** [comment]
I think I'd feel comfortable enough sharing the clientId/clientSecret with you. Its all set to Read-Only, and I can always change them afterwards, so it doesn't seem like a huge deal. What's the best way you think for me to share them?

**Comment by joeyagreco** [comment]
> I think I'd feel comfortable enough sharing the clientId/clientSecret with you. Its all set to Read-Only, and I can always change them afterwards, so it doesn't seem like a huge deal. What's the best way you think for me to share them?

Sounds good! Thanks

You can send them to my email with your league ID and year/s (if it's a multi-year league)

joeyagreco@gmail.com

**Comment by joeyagreco** [comment]
@tprusisz 

I received your info via email but am unable to load your league via the [yahoofantasy](https://github.com/mattdodge/yahoofantasy) library which is used in **leeger** on line 68 of **YahooLeagueLoader.py**:

`leagues = ctx.get_leagues(self.__NFL, currentYear)`

I snooped around a bit to see what the yahoofantasy library sees and looks like its not able to grab any leagues for 2022 with the info you sent me. (The `leagues` field in the attached image is empty)

Are you able to get past this locally? (i.e. are you getting past line 68 in **YahooLeagueLoader.py**?)


![image](https://user-images.githubusercontent.com/57115019/190053406-381cbf1c-615e-4f11-a88e-ec0ff3540264.png)

**Comment by tprusisz** [comment]
I want to say that I am making it further than that when running it, since my traceback begins to throw errors at line 83 of **YahooLeagueLoader.py**:

```
Traceback (most recent call last):
  File "C:\Users\tprus\Downloads\yahooLeagueLoaderExample2.py", line 8, in <module>
    league: League = yahooLeagueLoader.loadLeague()
  File "C:\Users\tprus\AppData\Local\Programs\Python\Python310\lib\site-packages\leeger\league_loader\YahooLeagueLoader.py", line 83, in loadLeague
    return self.__buildLeague(yahooLeagues)
  File "C:\Users\tprus\AppData\Local\Programs\Python\Python310\lib\site-packages\leeger\league_loader\YahooLeagueLoader.py", line 91, in __buildLeague
    years.append(self.__buildYear(yahooLeague))
  File "C:\Users\tprus\AppData\Local\Programs\Python\Python310\lib\site-packages\leeger\league_loader\YahooLeagueLoader.py", line 97, in __buildYear
    weeks = self.__buildWeeks(yahooLeague)
  File "C:\Users\tprus\AppData\Local\Programs\Python\Python310\lib\site-packages\leeger\league_loader\YahooLeagueLoader.py", line 118, in __buildWeeks
    if yahooMatchup.is_tied == 0:
AttributeError: 'Matchup' object has no attribute 'is_tied'
```

An error at that **ctx** line makes me think that it is not logging in properly for you. Does it not help if you include **yahooLeagueLoader.login(clientId, clientSecret)** before loading the league like below:

```
from leeger.league_loader import YahooLeagueLoader
from leeger.model.league import League

clientId = "XXXXXX"
clientSecret = "XXX"
yahooLeagueLoader = YahooLeagueLoader("XXXXXX", [2022], clientId=clientId, clientSecret=clientSecret, loginTimeoutSeconds=4)
yahooLeagueLoader.login(clientId=clientId, clientSecret=clientSecret)
league: League = yahooLeagueLoader.loadLeague()
```

**Comment by joeyagreco** [comment]
Ah I think I've identified the issue.

When the login happens, a redirect to localhost occurs, and Yahoo looks at the user that is logged in to authorize the app.

In this case, I am logged in to my Yahoo account, so when I approve this, it is setting me as the user.

This is reflected in the `user guid` field, which is the same when I run your league ID / credentials and mine.

I don't think there is a way around this unfortunately.

Perhaps we can set up a call sometime to debug this where you can run this on your machine and share screen.

**Comment by tprusisz** [comment]
Sure, we can try to find a time to do that. I don't have time today, but perhaps tomorrow afternoon would work.

**Comment by joeyagreco** [comment]
> Sure, we can try to find a time to do that. I don't have time today, but perhaps tomorrow afternoon would work.

Sounds good, I'll email you for details

**Comment by joeyagreco** [comment]
This issue was fixed in [version 1.3.2](https://github.com/joeyagreco/leeger/releases/tag/v1.3.2)
